### ISSUE TITLE
Applying ScoreV2 to Ranked/Loved marathons isn't working

### ISSUE THREAD

**Issue by cl8n** [ppy/osu-stable-issues#465]
this map https://osu.ppy.sh/beatmaps/2157291 is not having scorev2 applied to all the scores, you can see here 17-20 were in v2 and the rest are v1

![image](https://user-images.githubusercontent.com/9954349/83341985-70d8a480-a29e-11ea-8e4b-3a874447cd2e.png)

when you play the map, the scorev2 icon to added to the mods view on the top-right, but the scoring is still v1 if you didn't actually select the scorev2 mod in song select. this is with latest cuttingedge

**Comment by peppy** [comment]
Should be fixed in latest cutting edge (please confirm).

**Comment by cl8n** [comment]
looks like something about the deployment went wrong, the game tries to download an update on the main menu but it keeps getting 0518 (latest version before this). see confused comments on https://osu.ppy.sh/home/changelog/cuttingedge/20200518

**Comment by peppy** [comment]
Correct, external issues I'm following up on that are possibly out of our control.

**Comment by huuishuu** [comment]
As I've just managed to receive the latest update, I've briefly tested it out and it seems to be applying V2 correctly now. Tested nomod, NoFail and NFHTEZSO. With all the given scenarios, it's applying ScoreV2 as it should be.

**Comment by peppy** [comment]
Thanks. Closing this issue for now. We'll push these changes out to beta/stable this week.

**Comment by cl8n** [comment]
will earlier builds of the game be able to submit here still? and should I wait for stable's update to move the map back into loved?

**Comment by peppy** [comment]
Please wait a bit longer, will try and get beta/stable out today and add some backwards blocking logic.

**Comment by huuishuu** [comment]
![a6d0](https://user-images.githubusercontent.com/61030916/84408277-9d80aa80-ac14-11ea-9573-39f17a2874a6.png)
Reposting from the #osu-loved discussion on the osu!dev server - multiplayer doesn't seem to be applying the ScoreV2 fix, as denoted by Tary's score on the Stream Compilation 201X map.

![kOn8WIW](https://user-images.githubusercontent.com/61030916/84408413-c7d26800-ac14-11ea-8d1d-c7a4c46c83a5.png)

The fix works as intended in Single Player in every use-case scenario (except Auto) but not in Multiplayer.

**Comment by cl8n** [comment]
none of the online scores or replays have the V2 mod set, I think that's important for api users and UX on the leaderboard and score page. I'm assuming that if Auto fails to apply V2, that also means watching replays from the leaderboard in-game is still broken as well.

sorry this is a mess to fix, I'd usually try to test & report specific issues myself before moving map to Loved again, but there's not much I can test with a Pending map where score submission is involved

**Comment by peppy** [comment]
setting the v2 flag is not possible due to database limitations. fixing that will take weeks of migration time :(. need to figure a better way to handle that but for now that's how it is. might be best to hold off loving for the time being.

**Comment by huuishuu** [comment]
Would it be possible to have osu! block multiplayer scores in general on maps like this? Some sort of check that would prevent scores from submitting if it detects it's meant to be on ScoreV2? Though I'm not sure if something like that would work considering it has the possibility of breaking score submission in general.

**Comment by peppy** [comment]
hack on hack on hack
